jbmc/regression/jbmc-strings/CProverStringSetCharAt/Test.java

```diff
@@ -0,0 +1,18 @@
+import org.cprover.CProverString;
+
+public class Test
+{
+  public void test1(int index)
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    CProverString.setCharAt(sb, index, 'd');
+    assert sb.toString().equals("abc");
+  }
+
+  public void test2()
+  {
+    StringBuffer sb = new StringBuffer("abc");
+    CProverString.setCharAt(sb, -1, 'd');
```

Comment:
:pick: would be good to also have a test where the index is within the bounds and one where it exceeds them